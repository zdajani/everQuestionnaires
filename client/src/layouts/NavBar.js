import React, { Component } from 'react';
import { connect } from 'react-redux';
import { logout } from '../auth/actions';
import { NavLink } from 'react-router-dom';

export class NavBar extends Component {
   renderLogoutButton() {
     return (
       <button onClick={this.props.logout} className="nav-link logout-button btn btn-link">Logout</button>
     );
   }
   
   renderLoginLink() {
     return (
       <NavLink to="/login" className="nav-link">Login</NavLink>
     );
   }
   
  render() {
    return (
      <div className="container-fluid">
      <nav className="navbar navbar-expand-md navbar-dark fixed-top bg-dark">
        <div className="navbar-brand ttile">
          <h1 className="title">everQuestionnaires</h1>
        </div>
        <div className="nav-item">
          <NavLink to="/questionnaires" className="nav-link" activeClassName="active">
            Questionnaires
          </NavLink>
        </div>
        <div className="nav-item">
          <NavLink to="/questionnaires/new" className="nav-link" activeClassName="active">
            Create Questionnaire
          </NavLink>
        </div>
        <div className="nav-item">
          <NavLink to="/admin" className="nav-link" activeClassName="active">
            Admin Page
          </NavLink>
        </div>
        <div className="nav-item">
          {this.props.isAuthenthicated ? this.renderLogoutButton() : this.renderLoginLink()}
        </div>
      </nav>
    </div>
    )
  }
}


export default connect(null, { logout })(NavBar);
