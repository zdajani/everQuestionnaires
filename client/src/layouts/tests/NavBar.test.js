import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { NavBar } from '../NavBar';

const logoutSpy = jest.fn();
const isAuthenthicatedMock = {
  exp: 1234566,
  sub: 1
}

const setup = (isAuthenthicated = isAuthenthicatedMock) => {
  const component = shallow(
      <NavBar 
        logout={logoutSpy}
        isAuthenthicated={isAuthenthicated} />
    );
    
  return {
    logoutSpy,
    component
  };
}

describe('NavBar', () => {
  describe('when user logged in', () => {
    const { component, logoutSpy } = setup()
    it('renders self isAuthenthicated true props', () => {  
      expect(component.instance().props).toEqual({isAuthenthicated: isAuthenthicatedMock, logout: logoutSpy })
      expect(toJson(component)).toMatchSnapshot();
    });
    
    it('renders self with logout button', () => {  
      expect(toJson(component.find('button').first())).toMatchSnapshot();
    });
    
    it('calls logout on logout button', () => { 
      const {component, logoutSpy } = setup()
      component.find('button').first().simulate('click');
      expect(logoutSpy.mock.calls.length).toBe(1)
    });
  })
  
  describe('when user logged out', () => {
    const { component, logoutSpy } = setup(false)
    it('renders self with isAuthenthicated false props', () => {  
      expect(component.instance().props).toEqual({isAuthenthicated: false, logout: logoutSpy })
    });
    
    it('renders self with login link', () => {  
      expect(toJson(component)).toMatchSnapshot();
    });
  })
});