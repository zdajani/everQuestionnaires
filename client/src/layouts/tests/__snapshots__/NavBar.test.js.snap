// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar when user logged in renders self isAuthenthicated true props 1`] = `
<div
  className="container-fluid"
>
  <nav
    className="navbar navbar-expand-md navbar-dark fixed-top bg-dark"
  >
    <div
      className="navbar-brand ttile"
    >
      <h1
        className="title"
      >
        everQuestionnaires
      </h1>
    </div>
    <div
      className="nav-item"
    >
      <NavLink
        activeClassName="active"
        className="nav-link"
        to="/questionnaires"
      >
        Questionnaires
      </NavLink>
    </div>
    <div
      className="nav-item"
    >
      <NavLink
        activeClassName="active"
        className="nav-link"
        to="/questionnaires/new"
      >
        Create Questionnaire
      </NavLink>
    </div>
    <div
      className="nav-item"
    >
      <NavLink
        activeClassName="active"
        className="nav-link"
        to="/admin"
      >
        Admin Page
      </NavLink>
    </div>
    <div
      className="nav-item"
    >
      <button
        className="nav-link logout-button btn btn-link"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </nav>
</div>
`;

exports[`NavBar when user logged in renders self with logout button 1`] = `
<button
  className="nav-link logout-button btn btn-link"
  onClick={[Function]}
>
  Logout
</button>
`;

exports[`NavBar when user logged out renders self with login link 1`] = `
<div
  className="container-fluid"
>
  <nav
    className="navbar navbar-expand-md navbar-dark fixed-top bg-dark"
  >
    <div
      className="navbar-brand ttile"
    >
      <h1
        className="title"
      >
        everQuestionnaires
      </h1>
    </div>
    <div
      className="nav-item"
    >
      <NavLink
        activeClassName="active"
        className="nav-link"
        to="/questionnaires"
      >
        Questionnaires
      </NavLink>
    </div>
    <div
      className="nav-item"
    >
      <NavLink
        activeClassName="active"
        className="nav-link"
        to="/questionnaires/new"
      >
        Create Questionnaire
      </NavLink>
    </div>
    <div
      className="nav-item"
    >
      <NavLink
        activeClassName="active"
        className="nav-link"
        to="/admin"
      >
        Admin Page
      </NavLink>
    </div>
    <div
      className="nav-item"
    >
      <NavLink
        activeClassName="active"
        className="nav-link"
        to="/login"
      >
        Login
      </NavLink>
    </div>
  </nav>
</div>
`;
